tests/gnmi/helper.py

```diff
@@ -9,36 +10,37 @@
 GNMI_SERVER_START_WAIT_TIME = 15
 
 
-def gnmi_container(duthost):
-    global GNMI_CONTAINER_NAME
-    if len(GNMI_CONTAINER_NAME) == 0:
-        if get_image_type(duthost) == "public":
-            GNMI_CONTAINER_NAME = "telemetry"
-        else:
-            GNMI_CONTAINER_NAME = "gnmi"
-    return GNMI_CONTAINER_NAME
+@lru_cache(maxsize=None)
+class GNMIEnvironment(object):
+    def __init__(self, duthost):
+        cmd = "docker images | grep -w sonic-gnmi"
```

Comment:
Is it possible that we have both sonic-gnmi and sonic-telemetry docker image?

Reply:
Yes, it's possible, we enable both telemetry and gnmi in rules/config